bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java

```diff
@@ -254,9 +255,10 @@ public String toString() {
                 @Override
                 public Boolean call() throws Exception {
                     if (!isComplete() && null != maybeSendSpeculativeRead(heardFromHostsBitSet)) {
-                        if (LOG.isDebugEnabled()) {
-                            LOG.debug("Send speculative read for {}. Hosts heard are {}, ensemble is {}.",
-                                    this, heardFromHostsBitSet, ensemble);
+                        if (LOG.isInfoEnabled()) {
```

Comment:
Are we sure changing this from debug to info? At a high load cluster, you might have 5-10% or even more traffic do speculative reads to ensure predictable latency. This change will result in producing a lot of unnecessary logging. 

May I ask you a question? How useful this logging helps you guys at production?

Reply:
It's been pretty useful for us.  Perhaps we should simply bump logging to debug in the few classes where we want it. @jvrao thoughts?

Reply:
yeah. that would be ideal, you can configure your log4j properties to enable debug in PendingReadOp.